src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -130,6 +129,17 @@ internal static RuntimeGraph GetRuntimeGraph(IEnumerable targetFrameworks)
                            .Select(ToPackageDownloadDependency));
                 }
 
+                if (targetFrameworkInfo is IVsTargetFrameworkInfo3 targetFrameworkInfo3 && cpvmEnabled)
+                {
+                    if (targetFrameworkInfo3.CentralPackageVersions != null)
+                    {
+                        tfi.CentralPackageVersions.AddRange(
+                           targetFrameworkInfo3.CentralPackageVersions
+                               .Cast<IVsReferenceItem>()
+                               .Select(ToCentralPackageVersion).ToDictionary(cpv => cpv.Name));
+                    }
+                }
```

Comment:
> I don't know how I feel about this. If the nomination contains PackageVersions, even if the project is not CPVM enabled, it's still in the nomination.

We should ignore it. That's what we do from the command line.

Reply:
> I don't know how I feel about this. If the nomination contains PackageVersions, even if the project is not CPVM enabled, it's still in the nomination.

Yes this is correct. This is to avoid the Project System doing any business logic. Project System passes all the information to us. 




> If the goal is to avoid full restores when a PackageVersion changes for a project that is not CPVM enabled, could we transform the nomination as-is, but then strip out the PackageVersions at a later time? A method whose responsibility is to convert VS objects into `NuGet.ProjectModel` objects doesn't feel like the right place to put CPVM logic. This class is the VS equivalent of the DGSpec JSON reader. A JSON reader/parser isn't the right place to put CPVM business logic, and this class also doesn't feel like the right place for it.

This path is on the construction of the dgspec that will be sent to the restore. The dgspec needs to have the necessary information about central package versions. Why to pass the PackageVersions if not used? 

> Also why is it called CentralPackageVersions when that's not the name of the MSBuild item? but I know it's too late to change now if project system is already using it.

Yes, good point. The item name (PackageVersion) went through many re-naming cycles, one of them was CentralPackageVersion. The interface used that one. We can discuss how important is to change the name and evaluate if anything can be done. I will follow up on this, I do  not think that this is a PR blocker.

Reply:
> We should ignore it. That's what we do from the command line.

When we say "we should ignore it" , do you mean that  ProjectSystem should not pass us the information if the project does not have CPVM enabled?

Reply:
Should we ignore it in the "deserialization" where we convert VS objects into ProjectModel objects? Or should we "deserialize" as-is provided, and then fix-up in another method where it's more obvious what the business reason and business rules are?

Reply:
Project System can pass it, we should ignore it. 
I don't think it makes sense to process it. 
This already causes so many allocations.

Reply:
I agree.